pkg/module/file/file.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2025 Blindspot Software
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package file provides a dutagent module that transfers files between client and dutagent.
+package file
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/BlindspotSoftware/dutctl/pkg/module"
+)
+
+func init() {
+	module.Register(module.Record{
+		ID: "file",
+		New: func() module.Module {
+			return &File{}
+		},
+	})
+}
+
+// File permission constants.
+const (
+	DefaultDirPerm  = 0o755 // Default directory permissions
+	DefaultFilePerm = 0o644 // Default file permissions
+)
+
+// op represents the file operation.
+type op string
+
+const (
+	opUpload   op = "upload"
+	opDownload op = "download"
+)
+
+// File is a module that transfers files between client and dutagent.
+type File struct {
+	// Permission sets file permissions in octal format (e.g., "0644", "0755").
+	// Defaults: [DefaultFilePerm] for files, [DefaultDirPerm] for directories.
+	Permission string
+	// Operation pre-configures the operation type ("upload" or "download").
+	Operation string
+
+	// Source and Destination are optional config inputs (via YAML).
+	// If set, the command-line arg becomes the counterpart (dest or src respectively).
+	Source      string
+	Destination string
```

Comment:
```suggestion
	// Source is fixed file path that takes precedence over locations passed by arguments.
	// It is a path on the agent for downloads and a path on the client for uploads.
	// Paths set via this setting are less restricted then those passed via arguments, i.e. absolute paths are permitted.
	Source      string
	// Destination is fixed file path that takes precedence over locations passed by arguments.
	// It is a path on the client for downloads and a path on the agent for uploads.
	// Paths set via this setting are less restricted then those passed via arguments, i.e. absolute paths are permitted.
	Destination string
```

Reply:
- Comments should star with the name of the respective type, and it is worth putting a comment on each item for better experience in IDEs
- Every config is set via YAML, not needed to tell